api/groups/baseGroup.go

```diff
@@ -1,22 +1,68 @@
 package groups
 
 import (
+	"strings"
+
 	"github.com/gin-gonic/gin"
 	logger "github.com/multiversx/mx-chain-logger-go"
 	"github.com/multiversx/mx-chain-notifier-go/api/shared"
+	"github.com/multiversx/mx-chain-notifier-go/config"
 )
 
 var log = logger.GetOrCreate("api/groups")
 
 type baseGroup struct {
-	endpoints []*shared.EndpointHandlerData
+	endpoints             []*shared.EndpointHandlerData
+	additionalMiddlewares []gin.HandlerFunc
 }
 
 // RegisterRoutes will register all the endpoints to the given web server
 func (bg *baseGroup) RegisterRoutes(
-	ws gin.IRoutes,
+	ws *gin.RouterGroup,
+	apiConfig config.APIRoutesConfig,
 ) {
 	for _, handlerData := range bg.endpoints {
+		isOpen, isAuthEnabled := getEndpointStatus(ws, handlerData.Path, apiConfig)
+		if !isOpen {
+			log.Debug("endpoint is closed", "path", handlerData.Path)
+			continue
+		}
+
+		if isAuthEnabled {
+			ws.Use(bg.GetAdditionalMiddlewares()...)
```

Comment:
not sure if this is ok. If we'll add one more middleware, it will automatically become active if the auth is enabled